crates/nu-protocol/src/value/record.rs

```diff
@@ -234,35 +237,35 @@ impl Record {
         }
     }
 
-    /// Obtain an iterator to remove elements in `range`
-    ///
-    /// Elements not consumed from the iterator will be dropped
-    ///
-    /// ```rust
-    /// use nu_protocol::{record, Value};
-    ///
-    /// let mut rec = record!(
-    ///     "a" => Value::test_nothing(),
-    ///     "b" => Value::test_int(42),
-    ///     "c" => Value::test_string("foo"),
-    /// );
-    /// {
-    ///     let mut drainer = rec.drain(1..);
-    ///     assert_eq!(drainer.next(), Some(("b".into(), Value::test_int(42))));
-    ///     // Dropping the `Drain`
-    /// }
-    /// let mut rec_iter = rec.into_iter();
-    /// assert_eq!(rec_iter.next(), Some(("a".into(), Value::test_nothing())));
-    /// assert_eq!(rec_iter.next(), None);
-    /// ```
-    pub fn drain<R>(&mut self, range: R) -> Drain
-    where
-        R: RangeBounds<usize> + Clone,
-    {
-        Drain {
-            iter: self.inner.drain(range),
-        }
-    }
+    // /// Obtain an iterator to remove elements in `range`
+    // ///
+    // /// Elements not consumed from the iterator will be dropped
+    // ///
+    // /// ```rust
+    // /// use nu_protocol::{record, Value};
+    // ///
+    // /// let mut rec = record!(
+    // ///     "a" => Value::test_nothing(),
+    // ///     "b" => Value::test_int(42),
+    // ///     "c" => Value::test_string("foo"),
+    // /// );
+    // /// {
+    // ///     let mut drainer = rec.drain(1..);
+    // ///     assert_eq!(drainer.next(), Some(("b".into(), Value::test_int(42))));
+    // ///     // Dropping the `Drain`
+    // /// }
+    // /// let mut rec_iter = rec.into_iter();
+    // /// assert_eq!(rec_iter.next(), Some(("a".into(), Value::test_nothing())));
+    // /// assert_eq!(rec_iter.next(), None);
+    // /// ```
+    // pub fn drain<R>(&mut self, range: R) -> Drain
+    // where
+    //     R: RangeBounds<usize> + Clone,
+    // {
+    //     Drain {
+    //         iter: self.inner.drain(range)
+    //     }
+    // }
```

Comment:
Unfortunately, this is one of the methods that `EcoVec` doesn't yet support. I couldn't come up with an easy workaround, so I've just removed this method.